internal/buildinfo/version.go

```diff
@@ -8,17 +8,126 @@ import (
 	"time"
 
 	"github.com/blang/semver"
+	"github.com/loadsmart/calver-go/calver"
 	"github.com/superfly/flyctl/terminal"
 )
 
+type Version interface {
+	String() string
+	EQ(other Version) bool
+	Outdated(string) (bool, error)
+}
+
+type SemverVersion struct {
+	semver.Version
+}
+
+func ParseSemver(version string) (*SemverVersion, error) {
+	v, err := semver.ParseTolerant(version)
+	if err != nil {
+		return nil, err
+	}
+	return &SemverVersion{v}, nil
+}
+
+func (v SemverVersion) EQ(other Version) bool {
+	o, ok := other.(*SemverVersion)
+	if ok {
+		return v.Version.EQ(o.Version)
+	} else {
+		return false
+	}
+}
+
+func (v *SemverVersion) Outdated(other string) (bool, error) {
+	otherParsed, err := ParseSemver(other)
+	if err != nil {
+		return false, err
+	}
+	return v.LT(otherParsed.Version), nil
+}
+
+type CalverVersion struct {
+	calver.Version
+}
+
+const calverFormat = "YYYY.0M.0D.MICRO"
+
+func ParseCalver(version string) (*CalverVersion, error) {
+	v, err := calver.Parse(calverFormat, version)
+	if err != nil {
+		return nil, err
+	}
+	return &CalverVersion{*v}, nil
+}
+
+func (v CalverVersion) EQ(other Version) bool {
+	o, ok := other.(*CalverVersion)
+	if ok {
+		return v.Version.CompareTo(&o.Version) == 0
+	} else {
+		return false
+	}
+}
+
+func (v *CalverVersion) Outdated(other string) (bool, error) {
+	otherParsed, err := ParseCalver(other)
+	if err != nil {
+		return false, err
+	}
+	return v.CompareTo(&otherParsed.Version) < 0, nil
+}
+
+func ParseVersion(other string) (Version, error) {
+	version, err := ParseCalver(other)
+	if err != nil {
+		return ParseSemver(other)
+	} else {
+		return version, nil
+	}
+}
+
+type Versions []Version
+
+func (v Versions) Len() int {
+	return len(v)
+}
+
+func (s Versions) Less(i, j int) bool {
+	v1, ok := s[i].(*CalverVersion)
+	if ok {
+		v2, ok := s[j].(*CalverVersion)
+		if ok {
+			return v1.CompareTo(&v2.Version) == -1
+		} else {
+			// All calver versions are > semver versions.
+			return false
+		}
+	} else {
+		v1 := s[i].(*SemverVersion)
+		_, ok := s[j].(*CalverVersion)
+		if ok {
+			// All semver versions are < calver versions.
+			return true
+		} else {
+			v2 := s[j].(*SemverVersion)
+			return v1.Version.LT(v2.Version)
+		}
+	}
+}
+
+func (s Versions) Swap(i, j int) {
+    s[i], s[j] = s[j], s[i]
+}
```

Comment:
CI is complaining about formatting on `version.go`. It suggests running `gofmt -s`?